network/wsNetwork.go

```diff
@@ -1544,10 +1538,13 @@ func (wn *WebsocketNetwork) numOutgoingPending() int {
 	return len(wn.tryConnectAddrs)
 }
 
-var websocketDialer = websocket.Dialer{
-	Proxy:             http.ProxyFromEnvironment,
-	HandshakeTimeout:  45 * time.Second,
-	EnableCompression: false,
+// GetNetTransport returns an http.Transport that limits the number of connection
+// to comply with connectionsRateLimitingCount.
+func (wn *WebsocketNetwork) GetNetTransport() *http.Transport {
+	transport := http.DefaultTransport.(*http.Transport)
```

Comment:
Changing the "stock" transport isn't a good idea. ( as the data race suggests ).
instead, maybe something in the lines of 
```
func (wn *WebsocketNetwork) GetNetTransport() *http.Transport {
	// we want to create a copy of the stock transport object here.
	transport := *http.DefaultTransport.(*http.Transport)
	transport.DialContext = wn.GetDialer().DialContext
	return &transport
}
```